lib/spack/spack/filesystem_view.py

```diff
@@ -41,15 +41,26 @@
 _projections_path = '.spack/projections.yaml'
 
 
-def view_symlink(src, dst, view, spec):
+def view_symlink(src, dst, **kwargs):
+    # keyword arguments are irrelevant
+    # here to fit required call signature
     os.symlink(src, dst)
 
 
-def view_hardlink(src, dst, view, spec):
+def view_hardlink(src, dst, **kwargs):
+    # keyword arguments are irrelevant
+    # here to fit required call signature
```

Comment:
```suggestion
def view_hardlink(src, dst, view, spec=None):
```

Reply:
For the functions that discard the arguments, I'd rather keep it as kwargs to make clear more easily that they're irrelevant.